src/index.js

```diff
@@ -703,19 +731,36 @@ walletRouter.post('/send-tx',
   }
 
   if (inputs.length > 0) {
+    // In case the first input is a query command, we will overwrite the inputs array with results
     if (inputs[0].type === 'query') {
-      // First get sum of all outputs
-      const sumOutputs = outputs.reduce((acc, obj) => obj.value + acc, 0);
-      const utxos = getUtxosToFillTx(wallet, sumOutputs, inputs[0]);
-      if (!utxos) {
-        const response = {success: false, error: 'No utxos available for the query filter for this amount.'};
-        res.send(response);
-        lock.unlock(lockTypes.SEND_TX);
-        return;
+      const treatedInputs = []
+
+      // We need to fetch UTXO's for each token on the "outputs"
+      for (const element of tokens) {
+        const [tokenUid, tokenObj] = element
+
+        const queryOptions = {
+          ...inputs[0],
+          token: tokenUid
+        }
+        const utxos = getUtxosToFillTx(wallet, tokenObj.amount, queryOptions);
+        if (!utxos) {
+          const response = {
+            success: false,
+            error: 'No utxos available for the query filter for this amount.',
+            token: tokenUid
+          };
+          res.send(response);
+          lock.unlock(lockTypes.SEND_TX);
+          return;
+        }
+
+        for (const utxo of utxos) {
```

Comment:
Why this change? We map `utxos` to `treatedInputs` then assign it to `inputs`, the same as we did before.

Reply:
Each token iterated may have multiple `utxos`. The `treatedInputs` variable is basically an intermediate to reduce them all to a single resulting array.

Only after all the output tokens have been iterated we overwrite the `inputs` variable with the final result. That was the most concise solution I found.
Do you have an alternative suggestion?

Reply:
Didn't see that the inputs assignment was on another context. 
But i guess we could remove the `treatedInputs` anyway with:
```js
      // init inputs as empty array then
      inputs = inputs.concat(utxos.map((utxo) => {
        return {txId: utxo.tx_id, index: utxo.index};
      }));
```